### PR TITLE
Ksamarino registratūros langelis

### PR DESCRIPTION
**Pull request by AArlauskas**
Added registration screen and logic. SQLite

### CODE REVIEW

**Review by tmaciulis22 — CHANGES_REQUESTED** [review]
Resolve conflicts

File: app-android-xamarin/Moodis/Moodis/Resources/layout/activity_register.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <TextView
+        android:id="@+id/header"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Register"
```
Comment (Pilkas-git) [inline app-android-xamarin/Moodis/Moodis/Resources/layout/activity_register.xml]: You already created the string just forgot to use it "@string/register_text"

**Review by Pilkas-git — APPROVED** [review]
1 small change.

File: app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterActivity.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+using Android.Runtime;
+using Android.Views;
+using Android.Widget;
+using Moodis.Constants.Enums;
+using Moodis.Extensions;
+using Moodis.Feature.SignIn;
+
+namespace Moodis.Feature.Register
+{
+    [Activity(Label = "Register Activity")]
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterActivity.cs]: just `Register`

File: app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterActivity.cs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                else
+                {
+                    var regex = new Regex(@"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[#$^+=!*()@%&]).{8,15}$");
+                    if(regex.IsMatch(passwordInput.Text))
+                    {
+                        var response = await registerViewModel.AddUser(usernameInput.Text, passwordInput.Text);
+                        if(response == Response.OK)
+                        {
+                            Toast.MakeText(this, Resource.String.user_created, ToastLength.Short).Show();
+                            StartActivity(new Android.Content.Intent(this, typeof(SignInActivity)));
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterActivity.cs]: Don't finish sign in activity, when register button(at sign in activity) is clicked. And here just finish the register activity without starting new sign in activity.

Comment (tmaciulis22) [reply app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterActivity.cs]: And you can also set the result as OK, and then OnActivityResult in SignInActivity you could show Toast message of successful registration

File: app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs [hunk 4]
```diff
@@ -113,6 +114,9 @@ private void InitButtonsAndInputs()
             registerButton.Click += (sender, e) =>
             {
                 //StartActivityForResult(new Android.Content.Intent(this, typeof(RegisterActivity)), REQUEST_CODE_REGISTER);
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs]: Either use the suggestion I made regarding activity flow in RegisterActivity and uncomment this code, or delete it

File: app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInViewModel.cs [hunk 5]
```diff
@@ -19,7 +20,7 @@ public bool Authenticate(string username, string password)
             FetchUserList();
 
             //TODO Change to Crypto.CalculateMD5Hash(password));
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInViewModel.cs]: Remove this comment

File: app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInViewModel.cs [hunk 6]
```diff
@@ -58,8 +59,7 @@ public async Task<Response> AuthenticateWithFace(string imagePath)
 
         private void FetchUserList()
         {
-            userList = new List<User>();
-            userList.Add(new User("juris", "@Komparchas1"));
+            userList = DatabaseModel.FetchData();
 
             //if (File.Exists(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + "/users.bin"))
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInViewModel.cs]: remove this comment

**Review by tmaciulis22 — CHANGES_REQUESTED** [review]
Some changes with activity flow, but all in all everything looks good. We should meet up and decide how our other DB entities should look.

File: app-android-xamarin/Moodis/Moodis/Database/DatabaseModel.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using Moodis.Feature.Login;
+
+namespace Moodis.Database
+{
+    class databaseModel
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Database/DatabaseModel.cs]: Class names start with capital letters

File: app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterActivity.cs [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+                }
+                else
+                {
+                    var regex = new Regex(@"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[#$^+=!*()@%&]).{8,15}$");
+                    if(regex.IsMatch(passwordInput.Text))
+                    {
+                        var response = await registerViewModel.AddUser(usernameInput.Text, passwordInput.Text);
+                        if(response == Response.OK)
+                        {
+                            Toast.MakeText(this, Resource.String.user_created, ToastLength.Short).Show();
```
Comment (tmaciulis22) [inline app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterActivity.cs]: No need to create toast message here, if you do it in sign in activity on successful result

**Review by tmaciulis22 — APPROVED** [review]
Minor changes

File: app-android-xamarin/Moodis/Moodis/Database/DatabaseModel.cs (hunk 7 above)
Comment (AArlauskas) [reply app-android-xamarin/Moodis/Moodis/Database/DatabaseModel.cs]: yeah, I know, but at first DatabaseModel was throwing naming errors and databaseModel was not, so I left it. After some tinkering I managed to fix it

File: app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterActivity.cs (hunk 8 above)
Comment (AArlauskas) [reply app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterActivity.cs]: Yeah. Simply pamiršau

**Pull request merged**